### PR TITLE
ci: optimize glint

### PR DESCRIPTION
**Pull request by daixiang0**
Signed-off-by: Long Dai <long0dai@foxmail.com>

For an explanation of how to fill out the fields, please see the relevant section
in [PULL_REQUESTS.md](https://github.com/envoyproxy/envoy/blob/main/PULL_REQUESTS.md)

Commit Message:

Use `git diff` to get all changed files.

Additional Description:
Risk Level: Low
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: tools/code_format/glint.sh [hunk 1]
```diff
@@ -49,7 +49,7 @@ check_trailing_whitespace () {
 # TODO(phlax): add hash/diff only filter for faster change linting
 #      this would also make it feasible to add as a commit/push hook
 find_text_files () {
-    git grep --cached -Il '' | grep -vE "$NOLINT_RE"
+    git diff --cached --name-only --diff-filter=ACRMTU main | grep -vE "$NOLINT_RE"
```
Comment (phlax) [inline tools/code_format/glint.sh:52]: i like this idea, but a few points:

- in ci it should do a full scan
- it should work against changes in any branch not necessarily main
- the current way doesnt search in binary files (not that this matters a huge amount but would be good to keep that)

Comment (daixiang0) [reply tools/code_format/glint.sh:52]: For point one and two, we can pass some value to achieve it.
For point three, `git diff` does not search in binary files I think.

Comment (daixiang0) [reply tools/code_format/glint.sh:52]: Hi, is there a value about branch and CI env or not, if not, I will add them.

Comment (daixiang0) [reply tools/code_format/glint.sh:52]: ping @phlax

**Comment by zuercher** [comment]
Is this work still ongoing?

**Comment by phlax** [comment]
> Is this work still ongoing?

im in two minds about it

in ci its better that it checks all files - and this only runs in ci atm

that said - if we add this as a github hook then it becomes very useful - so i was leaving it on the back-burner for now

**Comment by htuch** [comment]
@daixiang0 @phlax can we decide to move forward or close this out? It's been 41 days stale, best not to park things that aren't going to be resolved soon.

**Comment by daixiang0** [comment]
sure.

**Comment by phlax** [comment]
@daixiang0 i think your change is broadly good here i just wanted to do some other things before landing it - ill reopen once i get to it

**Pull request closed without merging**